src/exberry_int/exberry_integration.py

```diff
@@ -0,0 +1,250 @@
+    username: str
+    password: str
+    tradingApiUrl: str
+    adminApiUrl: str
+    apiKey: str
+    secret: str
+
+
+class OutboundPriority(Enum):
+    MARKET_RECONNECT = -1
+    DEQUEUED_MESSAGE = 0
+    NEW_MESSAGE = 1
+
+
+class MultiplePriorityQueue:
+    def __init__(self, priorities: Type[Enum]):
+        self.queue = asyncio.PriorityQueue()
+        self.priority_counters = dict([(prio, itertools.count()) for prio in priorities])
+
+    async def put(self, priority, item):
+        await self.queue.put([priority.value, next(self.priority_counters[priority]), item])
+
+    async def get(self):
+        _, _, item = await self.queue.get()
+        return item
+
+
+class ExberryIntegration:
+    """ Handles connection to Exberry along with sending and receiving messages """
+    def __init__(self, env: 'ExberryIntegrationEnv', logger: Logger):
+        self.outbound_queue = MultiplePriorityQueue(OutboundPriority)
+        self.session_started = asyncio.Event()
+        self.env = env
+        self.last_tracking_number = None # Optional[int]
+        self.logger = logger
+
+
+    def set_last_tracking_number(self, tracking_number: int):
+        """
+        Set the tracking number that the integration will use to request an order book
+        stream starting from
+        """
+        self.last_tracking_number = tracking_number
+
+
+    async def start_session(self):
+        """ Indicate that the integration has established an Exberry websocket session """
+        self.session_started.set()
+
+
+    async def subscribe_to_order_book_depth(self):
+        """ Enqueues a message to be sent that will subscribe to Exberry's OrderBook stream """
+        data = { 'trackingNumber': self.last_tracking_number } if self.last_tracking_number else {}
+        subscription_request = {
+            'q': Endpoints.OrderBookDepth,
+            'sid': 0,
+            'd': data
+        }
+        await self.enqueue_outbound(subscription_request, OutboundPriority.MARKET_RECONNECT)
+
+
+    async def post_admin(self, data_dict: dict, endpoint: str) -> dict:
+        """ Post a message to the Exberry Admin API """
+        token = await self.__fetch_token()
+        async with ClientSession() as session:
+            self.logger.info(f'Integration ==> Exberry: POST {data_dict}')
+            async with session.post(f'{self.env.adminApiUrl}/{endpoint}',
+                                    json=data_dict,
+                                    headers={'Authorization': f'Bearer {token}'}) as resp:
+                return await resp.json()
+
+
+    async def enqueue_outbound(self, msg: dict, priority: OutboundPriority = OutboundPriority.NEW_MESSAGE):
+        """ Enqueue a message to be send through the Exberry websocket """
+        self.logger.info(f"Enqueuing outbound message: {msg} with priority {priority}")
+        await self.outbound_queue.put(priority, msg)
+
+
+    async def __dequeue_outbound(self) -> dict:
```

Comment:
What's the significance of the double leading underscore? Usually only one underscore is needed to signify _private_.

Reply:
Only that the naming convention I happened to click on when I googled used two underscore 😃 . I see now that a double underscore is something specific related to name-mangling when subclassing, I will change.